src/query/service/src/interpreters/interpreter_copy_v2.rs

```diff
@@ -49,6 +54,95 @@ impl CopyInterpreterV2 {
         Ok(CopyInterpreterV2 { ctx, plan })
     }
 
+    async fn filter_duplicate_files(
+        &self,
+        force: bool,
+        table_info: &StageTableInfo,
+        catalog_name: &str,
+        database_name: &str,
+        table_name: &str,
+        files: &[String],
+    ) -> Result<(u64, BTreeMap<String, TableCopiedFileInfo>)> {
+        let catalog = self.ctx.get_catalog(catalog_name)?;
+        let tenant = self.ctx.get_tenant();
+        let table = catalog
+            .get_table(&tenant, database_name, table_name)
+            .await?;
+        let table_id = table.get_id();
+        let req = GetTableCopiedFileReq {
+            table_id,
+            files: files.to_owned(),
+        };
+        let mut file_map = BTreeMap::new();
+
+        if !force {
+            let resp = catalog.get_table_copied_file_info(req).await?;
+            for file in files.iter() {
+                let stage_file = list_files(&self.ctx, &table_info.stage_info, file, "").await?;
+                if stage_file.is_empty() {
+                    continue;
+                }
+
+                let stage_file = &stage_file[0];
+
+                if let Some(file_info) = resp.file_info.get(file) {
+                    // No need to copy the file again if etag is_some and match.
+                    if stage_file.etag.is_some() && stage_file.etag == file_info.etag {
+                        continue;
+                    }
+
+                    if file_info.content_length == stage_file.size
+                        && file_info.last_modified == Some(stage_file.last_modified)
+                    {
```

Comment:
When do we fall in this case?

Reply:
If the ETag is the same, does the check is not need?

Reply:
> If the ETag is the same, does the check is not needed?

yes, now the logic is:

1. compare etag, if etag is_some and matched, ignore copy this file.
2. compare content-length and last_modified, if matched, ignore copy this file.

Reply:
Do not get the point:
If the etag is the same why need to check content_length&size again?